endpoint.go

```diff
@@ -777,7 +766,7 @@ func (ep *endpoint) Delete(force bool) error {
 
 	ep.releaseAddress()
 
-	return nil
+	return n.getEpCnt().DecEndpointCnt()
```

Comment:
The engine caller in this case anyways log if it sees an error. If we also log here then we will see redundant logs. In general if it is avoidable we should avoid logging in a library and allow the caller to make that decision.